File: tests/src/org.apache.pekko/pekko-protobuf-v3_3/1.0.1/src/test/scala/org_apache_pekko/pekko_protobuf_v3_3/MessageLiteToStringTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_apache_pekko.pekko_protobuf_v3_3

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class MessageLiteToStringTest {
  @Test
  def generatedMessageLiteToStringUsesReflectivePrinter(): Unit = {
    val message: GeneratedMessageLiteSerializedFormProbe = GeneratedMessageLiteSerializedFormProbe.getDefaultInstance

    val debugString: String = message.toString

    assertThat(debugString).startsWith("# ")
    assertThat(debugString).contains("GeneratedMessageLiteSerializedFormProbe")
  }
}

File: tests/src/org.apache.kafka/kafka-clients/4.2.0/src/test/java/org_apache_kafka/kafka_clients/MessageLiteToStringTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_apache_kafka.kafka_clients;

import org.apache.kafka.shaded.com.google.protobuf.TextFormat;
import org.apache.kafka.shaded.io.opentelemetry.proto.profiles.v1experimental.ProfilesData;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MessageLiteToStringTest {

    @Test
    void printsGeneratedTelemetryMessage() {
        ProfilesData data = GeneratedMessageV3Test.sampleProfilesData();

        String printed = TextFormat.printer().printToString(data);
        String simpleString = data.toString();

        assertThat(printed).contains("schema_url");
        assertThat(simpleString).contains("resource_profiles");
    }
}
